Services/Certificate/classes/Placeholder/Values/class.ilStudyProgrammePlaceholderValues.php

```diff
@@ -1,47 +1,21 @@
-<?php
+<?php declare(strict_types=1);
+
 /* Copyright (c) 1998-2018 ILIAS open source, Extended GPL, see docs/LICENSE */
 
 class ilStudyProgrammePlaceholderValues implements ilCertificatePlaceholderValues
 {
-    /**
-     * @var ilDefaultPlaceholderValues
-     */
-    private $defaultPlaceHolderValuesObject;
-
-    /**
-     * @var ilLanguage|null
-     */
-    private $language;
+    private ?ilDefaultPlaceholderValues $defaultPlaceHolderValuesObject;
+    private ?ilLanguage $language;
+    private ?ilCertificateObjectHelper $objectHelper;
+    private ?ilCertificateParticipantsHelper $participantsHelper;
+    private ?ilCertificateUtilHelper $ilUtilHelper;
```

Comment:
These types are not optional. Only the passed constructor arguments are optional.